fondat/aws/bedrock/resources/versions.py

```diff
@@ -0,0 +1,71 @@
+"""Resource for managing agent versions."""
+
+from collections.abc import Iterable
+from typing import Any, Mapping
+
+from fondat.aws.client import Config
+from fondat.pagination import Cursor
+from fondat.resource import resource
+from fondat.security import Policy
+
+from ..clients import agent_client
+from ..decorators import operation
+from ..pagination import decode_cursor, paginate
+
+
+@resource
+class VersionsResource:
+    """
+    Resource for managing agent versions.
+    Provides access to list versions and retrieve specific version details.
+    """
+
+    __slots__ = ("_agent_id", "config_agent", "policies")
+
+    def __init__(
+        self,
+        agent_id: str,
+        *,
+        config_agent: Config | None = None,
+        policies: Iterable[Policy] | None = None,
+    ):
+        self._agent_id = agent_id
+        self.config_agent = config_agent
+        self.policies = policies
+
+    @operation(method="get", policies=lambda self: self.policies)
+    async def get(self, *, max_results: int | None = None, cursor: Cursor | None = None) -> Any:
+        """
+        List all versions of the agent.
```

Comment:
I am wondering if there's a way to abstract the notion of versions (as well as aliases) so that it can apply to agents, flows and prompts (no alias for prompts).
This might make it easier to operate on a version object rather than having to create special purpose methods like `get_flow_version`. Thoughts ?